src/test/java/io/xstefank/wildfly/bot/PRDirectoriesVerificationTest.java

```diff
@@ -0,0 +1,180 @@
+        mockedContext = MockedGHPullRequestProcessor.builder(gitHubJson.id())
+                .prFile(".github/wildfly-bot.yml")
+                .repoDirectories(Set.of("src"));
+        given().github(mocks -> {
+                    Util.mockContext(mocks, mockedContext);
+                    GHRepository repo = mocks.repository(TestConstants.TEST_REPO);
+                    GHContent mockGHContent = mock(GHContent.class);
+                    when(repo.getFileContent(".github/" + RuntimeConstants.CONFIG_FILE_NAME, gitHubJson.commitSHA())).thenReturn(mockGHContent);
+                    when(mockGHContent.read()).thenReturn(IOUtils.toInputStream("""
+                                    wildfly:
+                                      rules:
+                                        - id: "id"
+                                          directories: [src]""",
+                            "UTF-8"));
+
+                })
+                .when().payloadFromString(gitHubJson.jsonString())
+                .event(GHEvent.PULL_REQUEST)
+                .then().github(mocks -> {
+                    GHRepository repo = mocks.repository(TestConstants.TEST_REPO);
+                    Mockito.verify(repo).getDirectoryContent("src");
+                    Mockito.verify(repo).createCommitStatus(gitHubJson.commitSHA(),
+                            GHCommitState.SUCCESS, "", "Valid", "Configuration File");
+                });
+    }
+
+    @Test
+    public void nonExistingDirectoryTest() throws IOException {
+        mockedContext = MockedGHPullRequestProcessor.builder(gitHubJson.id())
+                .prFile(".github/wildfly-bot.yml");
+        given().github(mocks -> {
+                    Util.mockContext(mocks, mockedContext);
+                    GHRepository repo = mocks.repository(TestConstants.TEST_REPO);
+                    GHContent mockGHContent = mock(GHContent.class);
+                    when(repo.getFileContent(".github/" + RuntimeConstants.CONFIG_FILE_NAME, gitHubJson.commitSHA())).thenReturn(mockGHContent);
+                    when(mockGHContent.read()).thenReturn(IOUtils.toInputStream("""
+                                    wildfly:
+                                      rules:
+                                        - id: "id"
+                                          directories: [src]""",
+                            "UTF-8"));
+
+                })
+                .when().payloadFromString(gitHubJson.jsonString())
+                .event(GHEvent.PULL_REQUEST)
+                .then().github(mocks -> {
+                    GHRepository repo = mocks.repository(TestConstants.TEST_REPO);
+                    Mockito.verify(repo).getDirectoryContent("src");
+                    Mockito.verify(repo).createCommitStatus(gitHubJson.commitSHA(),
+                            GHCommitState.ERROR, "", "Rule is missing an id or multiple rules have the same id.", "Configuration File");
```

Comment:
this doesn't make sense, custom message needed

Reply:
I think we can instead log all problems into the comment as is the case with format?